### ISSUE TITLE
Backslashes to be escaped for (projectile-project-root) in (projectile-codesearch-search)

### ISSUE THREAD

**Issue by sandos** [abingham/codesearch.el#19]
When running projectile-codesearch-search I get no results. After playing around a bit in elisp I came to the conclusion that backslashes are not escaped properly. I actually needed to double-escape them when eval-ing from a buffer like so:

> (codesearch-search "static" "c:\\\\\\\\cm\\\\\\\\projects\\\\\\\\550")

Whereas 
> (codesearch-search "static" "c:\\\\\\cm\\\\\\projects\\\\\\550")
> (codesearch-search "static" "c:\\\\cm\\\\projects\\\\550")
> (codesearch-search "static" "c:\cm\projects\550")

are either invalid or returns nothing.

**Comment by abingham** [comment]
Hmmm...I don't have a windows machine where I can easily investigate this (since I'm guessing this is largely going to be an issue on windows). Have you found any promising approaches?

You might try calling `shell-quote-argument` on `file-pattern` at `codesearch.el` line 180. This is a bit of a guess on my part, but I'd be curious to know if it makes your problem go away.

**Comment by sandos** [comment]
I tried your suggestion, and it does not seem to work. And I am not a elisp hacker, at all. But this works for me:

```
       (start-file-process "csearch" buff codesearch-csearch "-f" (replace-regexp-in-string "\\\\" "\\\\\\\\" file-pattern) "-n" pattern)
```



This will break if there is a real escape in file-pattern. I am not sure how to solve that.

**Comment by sandos** [comment]
A slightly more worked one, that just might work on other systems:

```elisp
(defun codesearch-search (pattern file-pattern)
  "Search files matching FILE-PATTERN in the index for PATTERN."
  (interactive
   (list
    (read-string "Pattern: " (thing-at-point 'symbol))
    (read-string "File pattern: " ".*")))
  (let ((process-environment (copy-alist process-environment))
        (switch-to-visible-buffer t)
        (buff (get-buffer-create "*codesearch-results*"))
	(fpattern (if (or (string= system-type "windows-nt")
			  (string= system-type "ms-dos"))
		      (replace-regexp-in-string "/" "\\\\\\\\" file-pattern)
		    file-pattern)))
    (setenv "CSEARCHINDEX" (expand-file-name codesearch-csearchindex))
    (with-current-buffer buff
      (read-only-mode 0)
      (erase-buffer)
      (set-process-sentinel
       (start-file-process "csearch" buff codesearch-csearch "-f" fpattern "-n" pattern)
       (lambda (process event)
         (codesearch--make-filenames-clickable (process-buffer process)))
       ))
    (pop-to-buffer buff)))
```

**Comment by abingham** [comment]
Fixed in #20
